github/resource_github_organization_project.go

```diff
@@ -0,0 +1,116 @@
+package github
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	"github.com/google/go-github/github"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceGithubOrganizationProject() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceGithubOrganizationProjectCreate,
+		Read:   resourceGithubOrganizationProjectRead,
+		Update: resourceGithubOrganizationProjectUpdate,
+		Delete: resourceGithubOrganizationProjectDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"body": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"url": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceGithubOrganizationProjectCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*Organization).client
+	o := meta.(*Organization).name
+	n := d.Get("name").(string)
+	b := d.Get("body").(string)
+
+	options := github.ProjectOptions{
+		Name: n,
+		Body: b,
+	}
+
+	project, _, err := client.Organizations.CreateProject(context.TODO(), o, &options)
+	if err != nil {
+		return err
+	}
+	d.SetId(strconv.FormatInt(*project.ID, 10))
+
+	return resourceGithubOrganizationProjectRead(d, meta)
+}
+
+func resourceGithubOrganizationProjectRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*Organization).client
+	o := meta.(*Organization).name
+
+	projectID, err := strconv.ParseInt(d.Id(), 10, 64)
+	if err != nil {
+		return err
+	}
+
+	project, resp, err := client.Projects.GetProject(context.TODO(), projectID)
+	if err != nil {
+		if resp != nil && resp.StatusCode == 404 {
+			d.SetId("")
+			return nil
+		}
+		return err
+	}
+
+	d.Set("name", project.GetName())
+	d.Set("body", project.GetBody())
+	d.Set("url", fmt.Sprintf("https://github.com/orgs/%s/projects/%d", o, project.GetNumber()))
```

Comment:
Is there any reason we should be constructing the URL here instead of reading it from the API? It appears that [`html_url`](https://developer.github.com/v3/projects/#create-an-organization-project) is the attribute we could use?

Reply:
Ah yes, but it is not made available in the go-github library? See [Project](https://godoc.org/github.com/google/go-github/github#Project).